test-case/check-volume-levels.sh

```diff
@@ -0,0 +1,337 @@
+#
+
+# 0..1s channels 1-4 previous gains
+# 1..2s channels 1-4 different mute switches
+# 2..3s channels 1-4 all muted
+run_test_2 () {
+    arecord_helper "$ARECORD_WAV2" "$CAP_CHANNELS" 4 & arecordPID=$!
+    nap
+    cset_nom; cset_mute_diff1 "$CAP_CHANNELS"; nap
+    cset_mute; nap
+
+    dlogi "Waiting $arecordPID"
+    wait $arecordPID
+    dlogi "Ready."
+}
+
+#
+# Test #3, test mute switch preservation from previous to next
+#
+
+# 0..1s channels 1-4 muted
+# 1..2s channels 1-4  nominal gain
+# 2..3s different mute switches
+# 3..4s channels 1-4 nominal gain
+run_test_3 () {
+    arecord_helper "$ARECORD_WAV3" "$CAP_CHANNELS" 4 & arecordPID=$!
+    nap
+    cset_nom; nap
+    cset_mute_diff2 "$CAP_CHANNELS"; nap
+    cset_unmute
+
+    dlogi "Waiting $arecordPID"
+    wait $arecordPID
+    dlogi "Ready."
+}
+
+#
+# Helper functions
+#
+
+cset_nom () {
+    dlogi "Set nominal volume"
+    amixer cset name="$CAP_VOLUME" $VOL_NOM
+}
+
+cset_max () {
+    dlogi "Set maximum volume"
+    amixer cset name="$CAP_VOLUME" $VOL_MAX
+}
+
+cset_min () {
+    dlogi "Set minimum volume"
+    amixer cset name="$CAP_VOLUME" $VOL_MIN
+}
+
+cset_mutevol () {
+    dlogi "Set mute via volume"
+    amixer cset name="$CAP_VOLUME" $VOL_MUTE
+}
+cset_mute () {
+    dlogi "Set mute via switch"
+    amixer cset name="$CAP_SWITCH" off
+}
+
+cset_unmute () {
+    dlogi "Set unmute"
+    amixer cset name="$CAP_SWITCH" on
+}
+
+nap () {
+    dlogi "Sleeping"
+    sleep 1
+}
+
+cset_mute_diff1 () {
+    dlogi "Set switch pattern 1"
+    case "$1" in
+	4)
+	    amixer cset name="$CAP_SWITCH" off,on,on,off ;;
+	2)
+	    amixer cset name="$CAP_SWITCH" off,on ;;
+	1)
+	    amixer cset name="$CAP_SWITCH" off ;;
+    esac
+}
+
+cset_mute_diff2 () {
+    dlogi "Set switch pattern 2"
+    case "$1" in
+	4)
+	    amixer cset name="$CAP_SWITCH" on,off,off,on ;;
+	2)
+	    amixer cset name="$CAP_SWITCH" on,off ;;
+	1)
+	    amixer cset name="$CAP_SWITCH" on ;;
+    esac
+}
+
+cset_volume_diff1 () {
+    dlogi "Set volume pattern 1"
+    case "$1" in
+	4)
+	    amixer cset name="$CAP_VOLUME" $VOL_P10DB,$VOL_0DB,$VOL_M10DB,$VOL_M30DB ;;
+	2)
+	    amixer cset name="$CAP_VOLUME" $VOL_P10DB,$VOL_0DB ;;
+	1)
+	    amixer cset name="$CAP_VOLUME" $VOL_P10DB ;;
+    esac
+}
+
+cset_volume_diff2 () {
+    dlogi "Set volume pattern 2"
+    case "$1" in
+	4)
+	    amixer cset name="$CAP_VOLUME" $VOL_M10DB,$VOL_P10DB,$VOL_0DB,$VOL_M20DB ;;
+	2)
+	    amixer cset name="$CAP_VOLUME" $VOL_M10DB,$VOL_P10DB ;;
+	1)
+	    amixer cset name="$CAP_VOLUME" $VOL_M10DB ;;
+    esac
+}
+
+arecord_helper () {
+    dlogi "Capturing file $1"
+    timeout -k 5 30 arecord -D "$CAP_HW" -f $CAP_FORMAT -r $CAP_RATE -c "$2" -d "$3" "$1"
+}
+
+do_preparations () {
+
+    if [[ "$tplg" != *"nocodec"* ]]; then
+	# Return special value 2 with exit
+	dlogi "This test is executed only with nocodec topologies. Returning Not Applicable."
+	exit 2
+    fi
+
+    test -n "$(command -v octave)" || {
```

Comment:
Simpler and logging the location
```shell
type octave || {
   ...
}
```

Reply:
I didn't know that, it's very compact but not very intuitive. If you are OK to re-review I can change this.